datalad/cmd.py

```diff
@@ -141,27 +143,106 @@ def run_gitcommand_on_file_list_chunks(func, cmd, files, *args, **kwargs):
     return ''.join(out), ''.join(err)
 
 
-def kill_output(output):
-    """Helper for WitlessRunner to swallow all output and neither
-    relay it to the parent process's stdout, nor provide it as the
-    return value of WitlessRunner.run().
-    """
-    return b'', 0
+async def run_async_cmd(loop, cmd, protocol, stdin, **kwargs):
+    lgr.debug('Async run %s', cmd)
+
+    cmd_done = asyncio.Future(loop=loop)
+    factory = functools.partial(protocol, cmd_done)
+    proc = loop.subprocess_exec(
+        factory,
+        *cmd,
+        stdin=stdin,
+        # ask the protocol which streams to capture
+        stdout=asyncio.subprocess.PIPE if protocol.proc_out else None,
+        stderr=asyncio.subprocess.PIPE if protocol.proc_err else None,
+        **kwargs
+    )
+    transport = None
+    try:
+        lgr.debug('Launching process %s', cmd)
+        transport, protocol = await proc
+        lgr.debug('Waiting for process %i to complete', transport.get_pid())
+        await cmd_done
+    finally:
+        # protect against a crash whe launching the process
+        if transport:
+            transport.close()
+
+    return cmd_done.result()
+
+
+class WitlessProtocol(asyncio.SubprocessProtocol):
+
```

Comment:
Given this is still in draft mode, you're probably planning on updating this and `run_async_cmd` with docstrings.   I think it'd be good to mention the link you put in the pr description (https://pymotw.com/3/asyncio/subprocesses.html), since this appears to follow that code pretty closely.

Reply:
Yes. And will do.

Reply:
Done.